src/go/rpk/pkg/api/admin/api_cloud_storage.go

```diff
@@ -41,6 +42,25 @@ type TopicRecoveryStatus struct {
 	RecoveryRequest RecoveryRequestParams `json:"request"`
 }
 
+// CloudStorageStatus represents the status of a partition of a topic in the
+// cloud storage.
+type CloudStorageStatus struct {
+	CloudStorageMode      string `json:"cloud_storage_mode"`                    // The partition's cloud storage mode (one of: disabled, write_only, read_only, full and read_replica).
+	LastManifestUpload    int    `json:"ms_since_last_manifest_upload"`         // Delta in milliseconds since the last upload of the partition's manifest.
+	LastSegmentUpload     int    `json:"ms_since_last_segment_upload"`          // Delta in milliseconds since the last segment upload for the partition.
+	LastManifestSync      *int   `json:"ms_since_last_manifest_sync,omitempty"` // Delta in milliseconds since the last manifest sync (only present for read replicas).
+	TotalLogSize          int    `json:"total_log_size_bytes"`                  // Total size of the log for the partition (overlap between local and cloud log is excluded).
+	CloudLogSize          int    `json:"cloud_log_size_bytes"`                  // Total size of the addressable cloud log for the partition.
+	LocalLogSize          int    `json:"local_log_size_bytes"`                  // Total size of the addressable local log for the partition.
```

Comment:
Can these be suffixed with Bytes, or swap Size for Bytes? Size doesn't indicate unit so it may be confusing eventually.

Reply:
Swapped Size for Bytes